cmd/syncthing/main.go

```diff
@@ -719,9 +709,51 @@ func syncthingMain() {
 		mainSvc.Add(relaySvc)
 	}
 
+	// Start discovery
+
+	cachedDiscovery := discover.NewCachingMux()
+	mainSvc.Add(cachedDiscovery)
+
+	if cfg.Options().GlobalAnnEnabled {
+		insecure := cfg.Options().GlobalAnnInsecure
```

Comment:
Actually, scrap that.

A DSN defines where we connect and what our connection parameters are.
The URL at which we post gets constructed from parts of the DSN, which to be honest is just host, and path. Query string is the only parameter we need and will post, and that comes from the thing consuming the DSN not the DSN itself, so I think it's fine.